transportation-data-publishing/data_tracker/signal_pms_fulcrum.py

```diff
@@ -0,0 +1,424 @@
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def get_last_run(args, knack_records):
+    """Summary: get the most recent modified date from all knack
+    signal_pm object
+    
+    Args:
+        args (namespace): command line inputs
+        knack_records (knack object): a list of all knack pm records
+        with fulcrum id
+    
+    Returns:
+        
+    """
+    knack_records_data = knack_records.data
+
+    if args.last_run_date:
+        return last_run_date
+
+    else:
+
+        for record in knack_records_data:
+
+            last_run_date_item = max(
+                knack_records_data, key=lambda x: x["MODIFIED_DATE"]
+            )
+
+            return last_run_date_item["MODIFIED_DATE"]
+
+
+
+def map_knack_id_signal_id(signals_records, payloads):
+    """Summary
+    
+    Args:
+        signals_records (knack object): all signal records from knack
+        payloads (list of dictionary): map signal_id on pm payloads
+    
+    Returns:
+        DataFrame: the prepared DataFrame
+    """
+    payloads = pd.DataFrame.from_dict(payloads)
+
+    signal_records_df = pd.DataFrame.from_dict(signals_records.data)
+
+    signal_records_df = signal_records_df[["SIGNAL_ID", "id"]]
+    signal_records_df = signal_records_df.rename(columns={"id": "SIGNAL"})
+
+    signal_records_df["SIGNAL_ID"] = signal_records_df["SIGNAL_ID"].astype(str)
+
+    signal_id_mapped_payloads = payloads.merge(
+        right=signal_records_df, left_on="signal_id", right_on="SIGNAL_ID", how="left"
+    )
+
+    return signal_id_mapped_payloads
+
+def prepare_pm_payloads(last_run_date, pgrest_records, signal_records, knack_pm_records):
+    """Summary
+    #TODO optimize the comparison
+    
+    Args:
+        last_run_date (UNIX Timestamp):  
+        pgrest_records (knack object): Description
+        signal_records (knack object): Description
+        knack_pm_records (knack object): Description
+    
+    Returns:
+        list: pms in postgre that are not in 
+    
+    Deleted Parameters:
+        last_run_date_item (TYPE): Description
+    """
+
+    pgrest_records_df = pd.DataFrame.from_dict(pgrest_records)
+    pgrest_records_df["modified_date"] = pgrest_records_df["modified_date"].apply(
+        datetime_to_unix_timestamp
+    )
+    pgrest_records_df["PM_STATUS"] = "COMPLETED"
+
+    pgrest_records_df = map_knack_id_signal_id(signal_records, pgrest_records_df)
+    pgrest_records_df.columns = map(str.upper, pgrest_records_df.columns)
+    pgrest_records_df = pgrest_records_df.rename(columns={"ID": "id"})
+
+    pgrest_records_list = pgrest_records_df.to_dict(orient="records")
+
+    pm_payloads = []
+
+
+    for pgrest_record in pgrest_records_list:
+        modified_date = pgrest_record["MODIFIED_DATE"]
+
+        if modified_date >= last_run_date:
+            pm_payloads.append(pgrest_record)
+
+    return pm_payloads
+
+def replace_pm_records(postgre_records, knack_pm_records, signal_records):
+    """Summary
```

Comment:
You've done a good job documenting the other functions. Would you write a brief summary/return definitions for this?